### ISSUE TITLE
Guest users should not have access to search results from password-protected folders.

### ISSUE THREAD

**Issue by HundSimon** [lyc8503/onedrive-cf-index-ng#19]
This appears to be a minor security issue. 
As the title suggests, guest users can get file names through searching.
I don't think it is an expected function.
![图片](https://github.com/lyc8503/onedrive-cf-index-ng/assets/77706174/77649c57-2cae-4eaf-8925-891d2a1b36ef)
Private folder is protected by password
(apologize for using your website as an example)

**Comment by lyc8503** [comment]
Thanks for raising the issue, this is indeed a security issue inherited from upstream, I'll try to fix it afterwards.

> (apologize for using your website as an example)

Don't worry, [as the documentation says](https://ovi.swo.moe/docs/features/protected-folders#caveats), this password protection feature doesn't offer very strong protection and I didn't put very private data in it. I also recommend that everyone don't use this as your only means of security, and if you need to share private files, I recommend using OneDrive's official sharing feature to generate a private sharing link.

**Comment by lyc8503** [comment]
Hello, I took a look at this issue you mentioned.

So far I've found that the different versions of the OneDrive API don't return exactly the same thing: OneDrive for Business/Education doesn't return the path information of the search results directly in the API, which makes it impossible for me to filter the search results directly. It takes a second request to get the path to each search result, however multiple requests on the backend can result in long delays or timeouts.

So maybe this improvement is not possible for now, sorry. For the time being, this behavior will stay as a "feature". I'll update the docs to mention this issue for future users, thanks for the issue.
